sds/src/proximity_keywords/included_keywords.rs

```diff
@@ -0,0 +1,261 @@
+use crate::proximity_keywords::next_char_index;
+use regex_automata::Input;
+
+pub struct CompiledIncludedProximityKeywords {
+    pub look_ahead_character_count: usize,
+    pub keywords_pattern: super::ProximityKeywordsRegex,
+}
+
+impl CompiledIncludedProximityKeywords {
+    pub fn keyword_matches<'a>(&'a self, content: &'a str) -> IncludedKeywordSearch<'a> {
+        IncludedKeywordSearch {
+            keywords: self,
+            content,
+            start: 0,
+        }
+    }
+}
+
+pub struct IncludedKeywordSearch<'a> {
+    keywords: &'a CompiledIncludedProximityKeywords,
+    content: &'a str,
+    start: usize,
+}
+
+impl<'a> IncludedKeywordSearch<'a> {
+    pub fn skip_to(&mut self, start: usize) {
+        if start > self.start {
+            self.start = start;
+        }
+    }
+}
+
+impl<'a> Iterator for IncludedKeywordSearch<'a> {
+    type Item = usize;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        // TODO: use a custom cache for this regex (https://datadoghq.atlassian.net/browse/SDS-329)
+        let input = Input::new(self.content).range(self.start..).earliest(true);
+
+        if let Some(included_keyword_match) =
+            self.keywords.keywords_pattern.content_regex.search(&input)
+        {
+            // The next scan starts at the next character after the start of the keyword since
+            // multi-word keywords can overlap
+            self.start = next_char_index(self.content, included_keyword_match.start())
+                .unwrap_or(included_keyword_match.end());
+            Some(included_keyword_match.start())
+        } else {
+            None
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use crate::proximity_keywords::{
+        compile_keywords_proximity_config, CompiledIncludedProximityKeywords,
+        ProximityKeywordsRegex,
+    };
+    use crate::{Labels, ProximityKeywordsConfig};
+    use regex_automata::meta;
+
+    fn compile_keywords(lookahead: usize, keywords: &[&str]) -> CompiledIncludedProximityKeywords {
+        let (included, _) = compile_keywords_proximity_config(
+            &ProximityKeywordsConfig {
+                look_ahead_character_count: lookahead,
+                included_keywords: keywords.iter().map(|s| s.to_string()).collect(),
+                excluded_keywords: vec![],
+            },
+            &Labels::empty(),
+        )
+        .unwrap();
+        included.unwrap()
+    }
+
+    #[test]
+    pub fn test_included_keyword_search() {
+        let keywords = super::CompiledIncludedProximityKeywords {
+            look_ahead_character_count: 0,
+            keywords_pattern: ProximityKeywordsRegex {
+                content_regex: meta::Regex::new(r"\b(foo|bar)\b").unwrap(),
+                path_regex: meta::Regex::new(r"unused").unwrap(),
+            },
+        };
```

Comment:
I think it can be simplified a bit:
```suggestion
        let keywords = compile_keywords(4, &["foo", "bar"]);
```


but it also seems that this test overlap with `should_detect_on_any_keyword` so possibly we can get rid of it